byterestclient/__init__.py

```diff
@@ -32,14 +32,18 @@ def __init__(self, token=None, endpoint=None, identifier='byterestclient'):
             'User-Agent': '%s:%s' % (socket.getfqdn(), identifier)
         }
 
-    def request(self, method, path, data=None, *args, **kwargs):
+    def request(self, method, path, data=None, referer=None, *args, **kwargs):
         url = self.format_absolute_url(path)
         request_method = getattr(requests, method)
+        headers = self.headers.copy()
+
+        if referer:
+            headers.setdefault('Referer', referer)
```

Comment:
assuming you mean something what line of code in what module, it might be easier to grab this info from the frames right inside the `byterestclient`, instead of making sure all its client pass a `referrer`